lib/queue/producer.ts

```diff
@@ -127,18 +145,68 @@ export async function enqueue(
 	actionRequest: ActionRequestContract,
 	schedule?: ProducerOptionsSchedule,
 ): Promise<ActionRequestContract> {
+	return enqueueWithOptions(
+		logContext,
+		pool,
+		actor,
+		actionRequest,
+		{},
+		schedule,
+	);
+}
+
+/**
+ * @summary Enqueue an action request with options
+ * @function
+ *
+ * @param logContext - log context
+ * @param pool - database connection pool
+ * @param actor - actor id
+ * @param actionRequest - action request contract
+ * @param options : enqueueing options ( see EnqueueOptions )
+ * @param schedule - scheduling options
+ * @returns enqueued action request contract
+ */
+export async function enqueueWithOptions(
```

Comment:
Ok. I made it initially this way because the `enqueue` function already has an optional parameter `schedule?: ProducerOptionsSchedule,`. Adding another optional parameter ( `priority` ) would force us to send the unneeded one as `null` which I don't like. But I'll give it another go; I agree that having one function is cleaner.

Reply:
@ramirogm I agree with your idea of merging `ProducerOptionsSchedule` and `EnqueueOptions`. Both of these contain options that are passed to graphile, so merging `ProducerOptionsSchedule` into `EnqueueOptions` might be a clean solution. What do you think?

Reply:
Merged on https://github.com/product-os/jellyfish-worker/pull/1592/commits/59d184147af79d8e80d06152e8c08dd009dc01df